### PR TITLE
fix: replacing package name in Kotlin template

### PR DESCRIPTION
**Pull request by szymonrybczak**
Summary:
---------

Closes https://github.com/react-native-community/cli/issues/2205
CLI when replacing placeholders in init command is looking for return `"HelloWorld"`, but during template's conversation from Java to Kotlin the function was simplified, and scripting for editing template doesn't see this anymore. So right now we need to look for `=`.

Test Plan:
----------

1. Clone the repository and do all the required steps from the [Contributing guide](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md)
2. Run this command:

```sh
node /path/to/react-native-cli/packages/cli/build/bin.js init hello --package-name com.hello.hello
```
3. Android app should compile and launch correctly.

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

**Comment by szymonrybczak** [comment]
Not sure if we want to keep support for Java templates.

**Comment by szymonrybczak** [comment]
Also I updated the fixtures, to meet the current state of Java/Kotlin template.

**Review by cortinico — APPROVED** [review]

File: packages/cli/src/commands/init/__tests__/__snapshots__/editTemplate.test.ts.snap [hunk 1]
```diff
@@ -78,21 +80,3 @@ exports[`should edit template 3`] = `
 +   \\"/ios/ProjectNameTests/.gitkeep\\",
     \\"/node_modules\\","
 `;
-
-exports[`should edit template with custom title 1`] = `
```
Comment (thymikee) [inline packages/cli/src/commands/init/__tests__/__snapshots__/editTemplate.test.ts.snap:82]: let's bring back this snapshot

File: packages/cli/src/commands/init/__tests__/editTemplate.test.ts [hunk 2]
```diff
@@ -88,10 +98,18 @@ skipIfNode20('should edit template', async () => {
...
   ).toMatchSnapshot();
+
+  expect(snapshotDiff(oldCFile, newCFile, {contextLines: 1})).toMatchSnapshot();
+  expect(
+    snapshotDiff(oldXmlFile, newXmlFile, {contextLines: 1}),
+  ).toMatchSnapshot();
+  console.log({
+    fixtureTree,
+    transformedTree,
+  });
```
Comment (thymikee) [inline packages/cli/src/commands/init/__tests__/editTemplate.test.ts:112]: leftover
```suggestion
```

File: packages/cli/src/commands/init/__tests__/editTemplate.test.ts [hunk 3]
```diff
@@ -88,10 +98,18 @@ skipIfNode20('should edit template', async () => {
...
   expect(
     snapshotDiff(fixtureTree.map(slash), transformedTree.map(slash), {
```
Comment (thymikee) [inline packages/cli/src/commands/init/__tests__/editTemplate.test.ts:114]: fixtureTree likely needs to be sorted alphabetically, I suppose this is why snapshot differs between mac and linux

File: packages/cli/src/commands/init/__tests__/__snapshots__/editTemplate.test.ts.snap [hunk 4]
```diff
@@ -21,78 +21,115 @@ exports[`should edit template 1`] = `
...
++     114,
+      111,
+-     108,
+-     100,
++     106,
+      101,
+-     114,
++     99,
++     116,
+      78,"
```
Comment (thymikee) [inline packages/cli/src/commands/init/__tests__/__snapshots__/editTemplate.test.ts.snap:134]: what's this? 😅

File: packages/cli/src/commands/init/__tests__/editTemplate.test.ts [hunk 5]
```diff
@@ -112,26 +133,36 @@ skipIfNode20('should edit template with custom title', async () => {
     path.resolve(
       FIXTURE_DIR,
       'android',
+      'android-java',
       'com',
       PLACEHOLDER_NAME.toLowerCase(),
-      'Main.java',
+      'MainActivity.java',
     ),
-    'utf8',
```
Comment (thymikee) [inline packages/cli/src/commands/init/__tests__/editTemplate.test.ts:119]: can we take this back?

Comment (szymonrybczak) [reply packages/cli/src/commands/init/__tests__/editTemplate.test.ts:119]: fixed

**Review by thymikee — APPROVED** [review]

**Pull request merged**